src/Raven.Server/Documents/Revisions/RevisionsStorage.ManuallyDelete.cs

```diff
@@ -0,0 +1,187 @@
+            }
+        }
+    }
+
+    internal sealed class DeleteRevisionsByChangeVectorManuallyMergedCommand : MergedTransactionCommand<DocumentsOperationContext, DocumentsTransaction>
+    {
+        private readonly List<string> _cvs;
+        private readonly bool _shouldThrowIfChangeVectorsNotFound;
+
+        public DeleteRevisionsByChangeVectorManuallyMergedCommand(List<string> cvs, bool shouldThrowIfChangeVectorsNotFound)
+        {
+            _cvs = cvs;
+            _shouldThrowIfChangeVectorsNotFound = shouldThrowIfChangeVectorsNotFound;
+        }
+
+        protected override long ExecuteCmd(DocumentsOperationContext context)
+        {
+            context.DocumentDatabase.DocumentsStorage.RevisionsStorage.DeleteRevisionsByChangeVectorManuallyInternal(context, _cvs, _shouldThrowIfChangeVectorsNotFound);
+            return 1;
+        }
+
+        public override IReplayableCommandDto<DocumentsOperationContext, DocumentsTransaction, MergedTransactionCommand<DocumentsOperationContext, DocumentsTransaction>>
+            ToDto(DocumentsOperationContext context)
+        {
+            throw new NotImplementedException();
+        }
+    }
+
+    public Task DeleteRevisionsByDocumentIdManuallyAsync(List<string> ids, long maxDeletes)
+    {
+        if (ids == null || ids.Count == 0)
+            return Task.CompletedTask;
+
+        return _database.TxMerger.Enqueue(new DeleteRevisionsByDocumentIdManuallyMergedCommand(ids, maxDeletes));
+    }
+
+    private void DeleteRevisionsByDocumentIdManuallyInternal(DocumentsOperationContext context, string id, long maxDeletes, ref long remainingDeletes)
+    {
+        var result = new DeleteOldRevisionsResult();
+
+        using (DocumentIdWorker.GetSliceFromId(context, id, out Slice lowerId))
+        using (GetKeyPrefix(context, lowerId, out Slice prefixSlice))
+        {
+            var collectionName = GetCollectionFor(context, prefixSlice);
+            if (collectionName == null)
+            {
+                if (_logger.IsInfoEnabled)
+                    _logger.Info($"Tried to delete revisions for '{id}' but no revisions found.");
+                return;
+            }
+
+            if (context.DocumentDatabase.DocumentsStorage.RevisionsStorage.IsAllowedToDeleteRevisionsManually(collectionName.Name) == false)
+                throw new InvalidOperationException($"You are trying to delete revisions of '{id}' but it isn't allowed by its revisions configuration.");
+
+            var table = EnsureRevisionTableCreated(context.Transaction.InnerTransaction, collectionName);
+            var newEtag = _documentsStorage.GenerateNextEtag();
+            var changeVector = _documentsStorage.GetNewChangeVector(context, newEtag);
+
+            var lastModifiedTicks = _database.Time.GetUtcNow().Ticks;
+            var revisionsToDelete = GetAllRevisions(context, table, prefixSlice, remainingDeletes, skipForceCreated: false, result);
```

Comment:
Doesn't this materialize 1M revisions, for example?

Reply:
It yield 1 revision every time